web/server/codechecker_server/migrations/report/versions/75ae226b5d88_review_status_for_each_report.py

```diff
@@ -0,0 +1,157 @@
+    # this field we need to read file contents from the database. Since file
+    # contents may be too large, and we store them in memory during this
+    # migration, we iterate over files by chunks and update the records of
+    # "reports" table only in the files of the given chunk. The chunk size can
+    # be given by FILE_CHUNK_SIZE.
+    FILE_CHUNK_SIZE = 30
+
+    def decode_file_content(content):
+        return zlib.decompress(content).decode('utf-8', errors='ignore')
+
+    dialect = op.get_context().dialect.name
+
+    col_rs = sa.Column('review_status', sa.Enum(
+        'unreviewed',
+        'confirmed',
+        'false_positive',
+        'intentional',
+        name='review_status'), server_default='unreviewed', nullable=True)
+    col_rs_author = sa.Column(
+        'review_status_author', sa.String(), nullable=True)
+    col_rs_date = sa.Column(
+        'review_status_date', sa.DateTime(), nullable=True)
+    col_rs_is_in_source = sa.Column(
+        'review_status_is_in_source',
+        sa.Boolean(), server_default='0', nullable=True)
+    col_rs_message = sa.Column(
+        'review_status_message', sa.Binary(), nullable=True)
+
+    conn = op.get_bind()
+
+    if dialect == 'sqlite':
+        op.execute('PRAGMA foreign_keys=off')
+        with op.batch_alter_table('reports') as batch_op:
+            batch_op.add_column(col_rs)
+            batch_op.add_column(col_rs_author)
+            batch_op.add_column(col_rs_date)
+            batch_op.add_column(col_rs_is_in_source)
+            batch_op.add_column(col_rs_message)
+        op.execute('PRAGMA foreign_keys=on')
+
+        conn.execute("""
+            UPDATE reports
+            SET (review_status, review_status_author, review_status_date, review_status_message) =
+                (SELECT status, author, date, message FROM review_statuses WHERE bug_hash = reports.bug_id)
+        """)
+        conn.execute("""
+            UPDATE reports
+            SET review_status = 'unreviewed'
+            WHERE review_status IS NULL
+        """)
+    elif dialect == 'postgresql':
+        op.add_column('reports', col_rs)
+        op.add_column('reports', col_rs_author)
+        op.add_column('reports', col_rs_date)
+        op.add_column('reports', col_rs_is_in_source)
+        op.add_column('reports', col_rs_message)
+
+        conn.execute("""
+            UPDATE reports
+            SET (review_status, review_status_author, review_status_date, review_status_message) =
+                (SELECT status, author, date, message FROM review_statuses WHERE bug_hash = reports.bug_id)
+        """)
+        conn.execute("""
+            UPDATE reports
+            SET review_status = 'unreviewed'
+            WHERE review_status IS NULL
+        """)
+
+    source_code_comment_handler = SourceCodeCommentHandler()
+
+    files_with_report = conn.execute("""
+        SELECT reports.file_id, files.content_hash
+        FROM reports INNER JOIN files ON reports.file_id = files.id
```

Comment:
Filter the `reports` where `review_status` column is `NOT NULL`, so we not process files which do not have source code comments.